include/swift/AST/Evaluator.h

```diff
@@ -73,15 +73,30 @@ class PrettyStackTraceRequest : public llvm::PrettyStackTraceEntry {
 /// An llvm::ErrorInfo container for a request in which a cycle was detected
 /// and diagnosed.
 template <typename Request>
-struct RequestErrorInfo : public ErrorInfo<Request> {
-  const Request &request;
+struct CyclicalRequestError : 
+  public llvm::ErrorInfo<CyclicalRequestError<Request>> {
 public:
+  static char ID;
+  const Request &request;
+
+  CyclicalRequestError(const Request &request): request(request) {}
 
-  RequestErrorInfo(Request &request): request(request) {}
+  virtual void log(llvm::raw_ostream &out) const {
+    out << "Cycle detected:\n";
+    simple_display(out, request);
+    out << "\n";
+  }
 
-  Request &getRequest() { return request; }
+  virtual std::error_code convertToErrorCode() const {
```

Comment:
Having implemented a few ErrorInfos myself, I'm pretty sure this is actually useless and should not be a requirement. Returning a success code is probably not great, though—it'd be better to trap.